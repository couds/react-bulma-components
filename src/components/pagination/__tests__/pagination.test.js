import React from 'react';
import { fireEvent, render } from 'react-testing-library';
import { Pagination } from '..';

describe('Pagination component', () => {
  it('should have 3 pages and page 1 active', () => {
    const { asFragment } = render(<Pagination total={3} current={1} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should have 5 pages, page 5 active and display pages 3 to 5', () => {
    const { asFragment } = render(<Pagination total={5} current={5} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should not display page numbers', () => {
    const { asFragment } = render(<Pagination showPrevNext delta={0} total={5} current={5} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should not display Previous/Next buttons', () => {
    const { asFragment } = render(<Pagination delta={3} total={5} current={2} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('Next button should be disabled', () => {
    const { asFragment } = render(<Pagination total={3} current={3} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should not render if total pages equals 1', () => {
    const { asFragment } = render(<Pagination total={1} current={1} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should not render if current page is greater than total pages', () => {
    const { asFragment } = render(<Pagination total={2} current={3} />);
    expect(asFragment()).toMatchSnapshot();
  });
  it('should call to onChange event on click on navigation button', () => {
    const onChange = jest.fn();
    const { container } = render(<Pagination total={5} current={0} showPrevNext onChange={onChange} />);
    fireEvent.click(container.querySelector('a.pagination-next'));
    expect(onChange).toHaveBeenCalledTimes(1);
    fireEvent.click(container.querySelector('a.pagination-previous'));
    expect(onChange).toHaveBeenCalledTimes(2);
    fireEvent.click(container.querySelectorAll('a.pagination-link')[1]);
    expect(onChange).toHaveBeenCalledTimes(3);
  });
});
