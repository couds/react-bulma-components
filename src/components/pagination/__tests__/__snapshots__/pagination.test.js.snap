// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pagination component Next button should be disabled 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination"
    class="pagination"
  >
    <ul
      class="pagination-list"
    >
      <li
        class="previous"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
      <li>
        <a
          aria-label="Page 1"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li>
        <a
          aria-label="Page 2"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li>
        <a
          aria-label="Page 3"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          3
        </a>
      </li>
      <li
        class="next"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Pagination component should have 3 pages and page 1 active 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination"
    class="pagination"
  >
    <ul
      class="pagination-list"
    >
      <li
        class="previous"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
      <li>
        <a
          aria-label="Page 1"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li
        class="active"
      >
        <a
          aria-current="page"
          aria-label="Page 2 is your current page"
          class="pagination-link is-current"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li>
        <a
          aria-label="Page 3"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          3
        </a>
      </li>
      <li
        class="next"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Pagination component should have 5 pages, page 5 active and display pages 3 to 5 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination"
    class="pagination"
  >
    <ul
      class="pagination-list"
    >
      <li
        class="previous"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
      <li>
        <a
          aria-label="Page 1"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li>
        <a
          aria-label="Page 2"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li>
        <a
          aria-label="Page 3"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          3
        </a>
      </li>
      <li>
        <a
          aria-label="Page 4"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          4
        </a>
      </li>
      <li>
        <a
          aria-label="Page 5"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          5
        </a>
      </li>
      <li
        class="next"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Pagination component should not display Previous/Next buttons 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination"
    class="pagination"
  >
    <ul
      class="pagination-list"
    >
      <li
        class="previous"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
      <li>
        <a
          aria-label="Page 1"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li>
        <a
          aria-label="Page 2"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li
        class="active"
      >
        <a
          aria-current="page"
          aria-label="Page 3 is your current page"
          class="pagination-link is-current"
          role="button"
          tabindex="0"
        >
          3
        </a>
      </li>
      <li>
        <a
          aria-label="Page 4"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          4
        </a>
      </li>
      <li>
        <a
          aria-label="Page 5"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          5
        </a>
      </li>
      <li
        class="next"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Pagination component should not display page numbers 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination"
    class="pagination"
  >
    <ul
      class="pagination-list"
    >
      <li
        class="previous"
      >
        <a
          aria-disabled="false"
          class="pagination-previous"
          role="button"
          tabindex="0"
        >
          Previous
        </a>
      </li>
      <li>
        <a
          aria-label="Page 1"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li>
        <a
          aria-label="Page 2"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li
        class="break"
      >
        <a
          class="pagination-ellipsis"
          role="button"
          tabindex="0"
        >
          ...
        </a>
      </li>
      <li>
        <a
          aria-label="Page 4"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          4
        </a>
      </li>
      <li>
        <a
          aria-label="Page 5"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          5
        </a>
      </li>
      <li
        class="next"
      >
        <a
          aria-disabled="false"
          class="pagination-next"
          role="button"
          tabindex="0"
        >
          Next
        </a>
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Pagination component should not render if current page is greater than total pages 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination"
    class="pagination"
  >
    <ul
      class="pagination-list"
    >
      <li
        class="previous"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
      <li>
        <a
          aria-label="Page 1"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li>
        <a
          aria-label="Page 2"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          2
        </a>
      </li>
      <li
        class="next"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;

exports[`Pagination component should not render if total pages equals 1 1`] = `
<DocumentFragment>
  <nav
    aria-label="pagination"
    class="pagination"
  >
    <ul
      class="pagination-list"
    >
      <li
        class="previous"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
      <li>
        <a
          aria-label="Page 1"
          class="pagination-link"
          role="button"
          tabindex="0"
        >
          1
        </a>
      </li>
      <li
        class="next"
      >
        <a
          aria-disabled="false"
          class=""
          role="button"
          tabindex="0"
        />
      </li>
    </ul>
  </nav>
</DocumentFragment>
`;
